wvd-templates/DSC/Functions.ps1

```diff
@@ -273,41 +278,82 @@ function AuthenticateRdsAccount {
         }
     }
     catch {
-        $errMsg = "Windows Virtual Desktop Authentication Failed, Error:`n$($_ | Out-String)"
-        Write-Log -Error "$errMsg"
-        throw "$errMsg"
+        #This creates a new script scope so that these variables aren't included in this function scope since this function is being called using dot source notation.
+        & {
+            Set-StrictMode -Version Latest
+            
+            $innerExAsStr = ""
+            $numInnerExceptions = 0
+            if($_.Exception -is [System.AggregateException] -and $_.Exception.InnerExceptions)
+            {
+                $numInnerExceptions = $_.Exception.InnerExceptions.Count
+                foreach ($innerEx in $_.Exception.InnerExceptions)
+                {
+                    if($innerExAsStr.Length -gt 0)
+                    {
+                        $innerExAsStr += "`n"
+                    } 
+                    $innerExAsStr += $innerEx
+                }
+            }
+            
+            $errMsg = "Error authenticating Windows Virtual Desktop account, ServicePrincipal=" + $ServicePrincipal
+            $errMsg += " Error Details:`n$($_ | Out-String)"

```

Comment:
I see that you are using `$PSItem | Out-String` in function `SetTenantGroupContextAndValidate` vs `$_ | Out-String` here. Whats the diff ?